File: tp-arithmetic-java/src/main/java/fr/isima/filters/MandatoryMathsOperatorFilter.java
package fr.isima.filters;

import java.io.IOException;

import javax.servlet.Filter;
import javax.servlet.FilterChain;
import javax.servlet.FilterConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebFilter;

/**
 * Vérifie la présence ou non des parametres pour les opérations mathématiques
 * 
 * @author Benjamin Kuchcik
 *
 */
@WebFilter("/maths/*")
public class MandatoryMathsOperatorFilter implements Filter {

    @Override
    public void destroy() {
    }

    @Override
    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws IOException, ServletException {
        final String a = req.getParameter("a");
        final String b = req.getParameter("b");
        if (a == null && b == null) {
            throw new ServletException("Membres a et b obligatoires");
        } else if (a == null) {
            throw new ServletException("Membre a obligatoire");
        } else if (b == null) {
            throw new ServletException("Membre b obligatoire");
        }
        chain.doFilter(req, resp);
    }

    @Override
    public void init(FilterConfig arg0) throws ServletException {

    }

}

File: tp-arithmetic-kotlin/src/main/kotlin/fr/isima/filters/MandatoryMathsOperatorFilter.kt
package fr.isima.filters

import javax.servlet.*
import javax.servlet.annotation.WebFilter

/**
 * Vérifie la présence ou non des parametres pour les opérations mathématiques
 *
 * @author Benjamin Kuchcik
 */
@WebFilter("/maths/*")
class MandatoryMathsOperatorFilter : Filter {


    override fun doFilter(req: ServletRequest, resp: ServletResponse, chain: FilterChain) {
        val a = req.getParameter("a")
        val b = req.getParameter("b")
        when {
            a == null && b == null -> throw ServletException("Membres a et b obligatoires")
            a == null -> throw ServletException("Membre a obligatoire")
            b == null -> throw ServletException("Membre b obligatoire")
            else -> chain.doFilter(req, resp)
        }
    }

    override fun destroy() {}

    override fun init(arg0: FilterConfig) {}

}
